### PR TITLE
Completed task 6 - GET/api/articles/:articleId/comments

### PR DESCRIPTION
**Pull request by cls-c**
Added new endpoint GET /api/particles/:articleId/comments 
Added new endpoint description in endpoints.json documentation; Updated /api endpoint response 
Added new test cases for GET /api/particles/:articleId/comments endpoint

### CODE REVIEW

File: controller.js [hunk 1]
```diff
@@ -39,12 +40,22 @@ exports.getArticleWithID = async (req, res,next) => {
...
   }
-}
\ No newline at end of file
+};
+
+exports.getArticleComments = async (req, res, next) => {
+  try {
+    const { articleId } = req.params;
+    const associatedComments = await fetchArticleComment(articleId);
+    res.status("200").send({ article: associatedComments });
```
Comment (nc-bela-bertalan) [inline controller.js:57]: Comments should be sent back on a key of 'comments' instead of 'article'.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -181,3 +183,75 @@ describe("GET /api/articles", () => {
...
+describe("GET /api/articles/:articleid/comments", () => {
+  test("should return 200", () => {
+    return request(app).get("/api/articles/1/comments").expect(200);
+  });
+  test("should return an array of object that all the comments of the given article ", () => {
+    const articleId = 1
+    const count = commentData.reduce((counter, obj) => {
+      if (obj.article_id === articleId) counter += 1
+      return counter;
+    }, 0)
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:196]: This works here, but as we have very simple test data, it's okay to just count the comments for an article and use that number instead of writing all this code to get that number. Less code -> less chance for bugs.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -181,3 +183,75 @@ describe("GET /api/articles", () => {
...
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "Bad Request: ID provided has not been found."
+        );
+      });
+  });
+});
+
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:257]: Please add another test where it responds with status 200 and an empty array of comments.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -181,3 +183,75 @@ describe("GET /api/articles", () => {
...
+        expect(body.article.length).toEqual(count);
+      });
+  });
+  test("should return an array of object for the expected article id, each with the following keys: title, article_idtopic, author,created_at,votes,image_img_url,comment_count  ", () => {
+    const articleId = 1
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.article).toEqual(
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:210]: For the correct response object, please see my comment in your controller.

**Review by nc-bela-bertalan — COMMENTED** [review]
Hey Sabrina, please check my comments and refactor your code.

File: __tests__/app.test.js (hunk 3 above)
Comment (cls-c) [reply __tests__/app.test.js:257]: I have a few thoughts of this: the 404 test case  or returning a 404 is not the best practice for this endpoint. 
Technically this is not an error, it is not the endpoint's concern whether the id provided was valid or not. It will return a 200 with empty list, and 404 in most forums (Stackoverflow discussion) are not recommended. 
https://apihandyman.io/empty-lists-http-status-code-200-vs-204-vs-404/ 

It's function is to look up specifically for the article_id and return if there's an associated comment for it. In this case, it should be a 200. However I understand you can also look at it as invalid path given that the non-existent id is given as part of the path parameter. I have amended the code to reflect both cases: 
404 when article_id does not exist
200 when article_id exist, but no comments.
